glDemo/main.cpp

```diff
@@ -86,6 +100,41 @@ int main() {
 	// *** setup viewplane to the appropriate size
 	gluOrtho2D(-1.1f, 1.1f, -1.1f, 1.1f);
 
+
+	random_device rd;
+	engine = mt19937(rd());
+	range = uniform_real_distribution<float>(-1.0f, 1.0f);
+
+	random_device dr;
+	engine2 = mt19937(dr());
+	range2 = uniform_real_distribution<float>(-1.0f, 20.0f);
```

Comment:
You set the range for the point size from -1 to 20.  This ideally should be +1 to 20.